airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -103,10 +110,22 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
 
 
 def launch(source: Source, args: List[str]):
-    source_entrypoint = AirbyteEntrypoint(source)
+    source_entrypoint = AirbyteEntrypoint(source, args)
+
+    def excepthook(exctype, excvalue, exctraceback):
+        import traceback
+
+        sys.stderr.write("An uncaught exception occurred. Terminating...\n")
+        traceback.print_exception(exctype, excvalue, exctraceback)
+        exit()
```

Comment:
what error code should we expect on application exit? In case of exception it should be non-zero. Should we call exit here? Do we need to set our own except hook if we already using stdout / stderr logging writer?